docker/Dockerfile

```diff
@@ -39,28 +40,34 @@ RUN apt-get update \
   && mkdir -p -m 0700 /run/user/"${UID}" \
   && chown -R hope:hope /code /tmp /data /run/user/"${UID}"
 
+ENV PDM_PACKAGES=/packages
+ENV CODE=/code
+ENV PDM_NO_SELF=True
+ENV PDM_PROJECT=$PDM_PACKAGES
+ENV PYPACKAGES=$PDM_PACKAGES/__pypackages__/3.11
 ENV PYTHONPYCACHEPREFIX=/tmp/pycache \
-    VIRTUAL_ENV=/opt/venv \
-    POETRY_HOME=/opt/poetry \
-    PATH=/opt/venv/bin:/opt/poetry/bin:${PATH} \
+    PYTHONPATH=$PYPACKAGES/lib:$CODE:$PYTHONPATH \
+    PATH=$PYPACKAGES/bin:$PATH \
     XDG_RUNTIME_DIR=/run/user/"${UID}"
 
-WORKDIR /code
+WORKDIR $CODE
 
 COPY --from=waitforit /data/waitforit /usr/local/bin/waitforit
 
+RUN pip install pdm==2.15.2 &&\
```

Comment:
do we need to have pdm in the dist (from base stage)?
can't this be moved to builder stage, as previously?

Reply:
sure, good catch